openwisp_controller/connection/base/models.py

```diff
@@ -44,6 +44,13 @@ def clean(self):
         self._validate_connector_schema()
 
     def _get_connector(self):
+        if not hasattr(self, '_connectors'):
+            self._connectors = dict(app_settings.CONNECTORS)
+        try:
+            if self._connectors[self.credentials.connector] == 'SNMP':
+                return self.credentials.connector
+        except AttributeError:
+            pass
```

Comment:
the base connector should now know anything about classes who inherit it, it should not know about SNMP.
Why is this necessary at all? Shouldn't it be enough to just specify `_connector_field` on the Snmp class?

Reply:
The `DeviceConnection` class uses `update_strategy` as `_connector_field` but it's meaningless for SNMP, so this here overrides that, otherwise even if we use an SNMP credential, it will automatically select SSH from the update strategy. Do you think there is a better way to change it?

Reply:
Yes, I think there's a better solution.

Base should not know the internals of SSH or SNMP, but it can know that some connectors are suited to perform updates and others are not, so we can introduce a generic attribute called `has_update_strategy` or something similar, which indicates whether the code which looks for an update strategy has to be executed or not.

So the SSH connectors would have this attribute set to True while SNMP connectors would have it to False.

Is it clear?